pinot-core/src/main/java/org/apache/pinot/core/segment/processing/partitioner/ColumnValuePartitioner.java

```diff
@@ -0,0 +1,39 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.core.segment.processing.partitioner;
+
+import org.apache.pinot.spi.data.readers.GenericRow;
+
+
+/**
+ * Partitioner which extracts a column value as the partition
+ */
+public class ColumnValuePartitioner implements Partitioner {
+
+  private final String _columnName;
+
+  public ColumnValuePartitioner(String columnName) {
+    _columnName = columnName;
+  }
+
+  @Override
+  public String getPartition(GenericRow genericRow) {
```

Comment:
Is this intended for supporting time alignment?

What if the time column granularity is in seconds/hours while push frequency is DAY?

In that case, we may need to use `TransformationPartitioner`?

Reply:
Not particularly written for supporting time alignment, but can be used for that if values are to be used as is for partitioning.

If time column is seconds/hours, but we want to align by day, yes we can use TransformFunctionPartitioner.